### PR TITLE
Ideact error

### LINKED ISSUE #671: IDE act: Listener for errors in a file
This could be used in the case of #668.

### PR DESCRIPTION
**Pull request by jaakkonarhi**
# Description of the PR

+ Closes #671 and #668
```
     {
          "instruction": "fix errors",
          "component": "editor",
          "info": "please",
	  "assertClosed": [],
          "action": "errors",          
          "actionArguments": {
            "filePath": "/Football1/o1/football1/Match.scala",
	  }           
       },
```

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>e2e</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [ ] new <b>unit</b> tests created</li>
        <li>- [x] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [x] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>e2e</b> tests created</li>
        <li>- [x] all <b>e2e</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [x] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](https://github.com/Aalto-LeTech/intellij-plugin/blob/master/TESTING.md) or other relevant documentation on _your_ branch?

- [ ] Yes.
- [ ] Not yet. I will do it next.
- [x] Not relevant.

### CODE REVIEW

File: src/main/java/fi/aalto/cs/apluscourses/intellij/psi/PsiUtil.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  }
+
+  public static boolean psiSiblingHasErrors(@NotNull PsiElement element) {
+    return psiElementHasErrors(element.getPrevSibling()) || psiElementHasErrors(element.getNextSibling());
+  }
+
+  public static boolean psiElementHasErrors(@NotNull PsiElement element) {
+    return !SyntaxTraverser.psiTraverser(element).traverse().filter(PsiErrorElement.class).isEmpty();
+  }
+}
```
Comment (Taikelenn) [inline src/main/java/fi/aalto/cs/apluscourses/intellij/psi/PsiUtil.java:20]: Is this whole class ever used anywhere in the code

Comment (jaakkonarhi) [reply src/main/java/fi/aalto/cs/apluscourses/intellij/psi/PsiUtil.java:20]: it's for @stellatsv  to use later

Comment (Taikelenn) [reply src/main/java/fi/aalto/cs/apluscourses/intellij/psi/PsiUtil.java:20]: acknowledged

**Review by Taikelenn — APPROVED** [review]

**Review by stellatsv — APPROVED** [review]
Works well!

**Pull request merged**